pkg/combination/combination_test.go

```diff
@@ -35,25 +35,46 @@ var combinationTests = []struct {
 			err:          nil,
 		},
 	},
+	{
+		name:  "standard set of long args",
+		input: testdata.LongCombinationInput,
+		expected: expected{
+			combinations: testdata.LongCombinationOutput,
+			err:          nil,
+		},
+	},
 }
 
-func TestAll(t *testing.T) {
+func TestNext(t *testing.T) {
+	// Test 1: Uses testIterative()
 	for _, tt := range combinationTests {
 		t.Run(tt.name, func(t *testing.T) {
 			combinationStream := NewStream(
 				WithArgs(tt.input),
-				WithSolveAhead(),
 			)
-			got, err := combinationStream.All()
-			if !errors.Is(err, tt.expected.err) {
-				t.Fatal("error received while retreiving all combinations:", err)
+
+			var got []map[string]string
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			for {
+				next, err := combinationStream.Next(ctx)
+				if !errors.Is(err, tt.expected.err) {
+					t.Fatal("[TEST nextIterative()] error received while processing combination stream:", err)
+				}
+
+				if next == nil {
+					break
+				}
+
+				got = append(got, next)
 			}
-			require.ElementsMatch(t, got, tt.expected.combinations, "Combos generated incorrectly")
+			require.ElementsMatch(t, got, tt.expected.combinations, "[TEST nextIterative()] Combos generated incorrectly")
 		})
 	}
-}
 
-func TestNext(t *testing.T) {
+	// Test 2: Uses nextPreSolvedCombinations()
```

Comment:
I think my preference would be to avoid having to separate tests within a single top-level test spec. It seems like my suggestion in https://github.com/operator-framework/combo/pull/70/files#r795839125 is more applicable now, and you could also pass a `solveAhead` parameter as a struct field that the testing code can handle, e.g. `WithSolveAhead(tt.solveAhead)`.